integration/service/spatialindexer/s2indexer/indexer.go

```diff
@@ -0,0 +1,76 @@
+package s2indexer
+
+import (
+	"github.com/Telenav/osrm-backend/integration/service/poiloader"
+	"github.com/Telenav/osrm-backend/integration/service/spatialindexer"
+	"github.com/golang/geo/s2"
+	"github.com/golang/glog"
+)
+
+type s2Indexer struct {
+	cellID2PointIDs  map[s2.CellID][]spatialindexer.PointID
+	pointID2Location map[spatialindexer.PointID]spatialindexer.Location
+}
+
+// NewS2Indexer generates spatial indexer based on google s2
+func NewS2Indexer() *s2Indexer {
```

Comment:
`New()`. Also, the `s2Indexer` isn't exported, can it be used by caller?

Reply:
So far, I don't expect it be used by caller.  Caller's code will looks like

```go
indexer := NewS2Indexer.Build().Dump(xxx)
```
I will try to put `indexer` related algorithm in side this package.   Will try.

Reply:
<del>
> So far, I don't expect it be used by caller. Caller's code will looks like
> 
> ```go
> indexer := NewS2Indexer.Build().Dump(xxx)
> ```
> 
> I will try to put `indexer` related algorithm in side this package.
</del>